plugins/inputs/statsd/statsd.go

```diff
@@ -394,11 +403,13 @@ func (s *Statsd) udpListen() error {
 				log.Printf("E! Error READ: %s\n", err.Error())
 				continue
 			}
-			bufCopy := make([]byte, n)
-			copy(bufCopy, buf[:n])
+			b := s.bufPool.Get().(*bytes.Buffer)
+			b.Reset()
+			b.Write(buf[:n])
 
 			select {
-			case s.in <- bufCopy:
+			case s.in <- b.Bytes():
```

Comment:
Doesn't `b.Bytes()` copy the buffer?

Reply:
No it doesn't. It just returns the internal `[]byte` slice.

[source](https://golang.org/src/bytes/buffer.go?s=1990:2021#L42)
```Go
func (b *Buffer) Bytes() []byte { return b.buf[b.off:] }
```

Reply:
This looks dangerous (maybe what @danielnelson was getting at).
As you've indicated `b.Bytes()` returns the underlying byte slice. When you send that byte slice to a chan, the other end of the chan might not have processed it yet. But you're putting that buffer back in the pool, which means it's available for checkout & modification.

Reply:
Ah ! Good one, let me update the PR.

Reply:
Hmm .. this is a bit difficult that I thought. The parsing is happening in another goroutine. And it's not possible for me to retrieve the original byte buffer _after_ it has been consumed :cry: 

I think the overhead of doing this correctly is more than the original implementation itself. Open to what you guys think.

Reply:
I guess if you pass the bytes.Buffer through the channel it will make a copy?  What about storing fixed length buffers in the bufPool (does statsd even have a max length message)?

Reply:
Thanks. Updated it.